sql/analyzer/resolve_views.go

```diff
@@ -109,28 +110,57 @@ func resolveViews(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel R
 	})
 }
 
+// applyAsOfToView transforms the nodes in the view's execution plan to apply the asOf expression to every
+// individual table involved in the view.
 func applyAsOfToView(n sql.Node, a *Analyzer, asOf sql.Expression) (sql.Node, transform.TreeIdentity, error) {
-	a.Log("applying AS OF clause to view definition")
-
 	return transform.Node(n, func(n sql.Node) (sql.Node, transform.TreeIdentity, error) {
-		urt, ok := n.(*plan.UnresolvedTable)
-		if !ok {
+		switch nn := n.(type) {
+		case *plan.Union:
```

Comment:
You should just use transform.NodeWithOpaque here instead of special casing particular node types

I think subqueries will have the same problem as unions, you should add a test

Reply:
sweet! thanks for pointing out `NodeWithOpaque`.